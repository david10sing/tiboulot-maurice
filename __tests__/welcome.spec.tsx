/* eslint-disable @typescript-eslint/no-unused-vars */

import 'react-native';
import React, { ReactElement } from 'react';

// Note: test renderer must be required after react-native.
import * as eva from '@eva-design/eva';
import { render, fireEvent } from '@testing-library/react-native';
import customTheme from '../src/tiboulot-maurice-theme.json';
import TiBoulotAppProvider from '../src/components/appProvider';
import WelcomeScreen, { WelcomeScreenProps } from '../src/screens/welcome';

/* eslint-enable @typescript-eslint/no-unused-vars */

jest.useFakeTimers();

const RenderedScreen = (props: WelcomeScreenProps): ReactElement => {
  return (
    <TiBoulotAppProvider {...eva} theme={{ ...eva.light, ...customTheme }}>
      <WelcomeScreen {...props} />
    </TiBoulotAppProvider>
  );
};

describe('rendering of the welcome screen', () => {
  it('should render a welcome screen', async () => {
    const { getByText, getByTestId, toJSON } = render(<RenderedScreen />);

    const welcomeTitle = getByText('Welcome to TiBoulot');
    expect(welcomeTitle).toBeTruthy();
    const welcomeSubtitle = getByText('Multiple services right in your pocket');
    expect(welcomeSubtitle).toBeTruthy();
    const signInBtn = getByText('Sign In');
    expect(signInBtn).toBeTruthy();
    const signUpBtn = getByText('Sign Up');
    expect(signUpBtn).toBeTruthy();
    const socialDivider = getByTestId('socialDivider');
    expect(socialDivider).toBeTruthy();
    const facebookBtn = getByTestId('facebookBtn');
    expect(facebookBtn).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });
});

describe('behaviours on the welcome screen', () => {
  it('should navigate to the sign in screen', () => {
    const onSignInMock = jest.fn();
    const { getByTestId } = render(<RenderedScreen onSignIn={onSignInMock} />);

    const signInBtn = getByTestId('signInBtn');
    fireEvent(signInBtn, 'onPress');
    expect(onSignInMock).toHaveBeenCalled();
  });

  it('should navigate to the sign in screen', () => {
    const onSignUpMock = jest.fn();
    const { getByTestId } = render(<RenderedScreen onSignUp={onSignUpMock} />);

    const signUpBtn = getByTestId('signUpBtn');
    fireEvent(signUpBtn, 'onPress');
    expect(onSignUpMock).toHaveBeenCalled();
  });
});
