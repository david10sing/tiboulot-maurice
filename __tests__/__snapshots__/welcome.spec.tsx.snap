// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the welcome screen should render a welcome screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={false}
      appearance="default"
      focusable={true}
      level="1"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
          },
          Array [
            Object {
              "flex": 1,
              "position": "relative",
            },
            Object {
              "backgroundColor": "#F7F9FC",
            },
            Object {
              "padding": 20,
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            appearance="default"
            level="1"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                Array [
                  Object {
                    "borderRadius": 14,
                    "padding": 20,
                    "width": "100%",
                  },
                  Object {
                    "backgroundColor": "#FFF",
                    "elevation": 2,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 4,
                  },
                ],
              ]
            }
          >
            <View
              appearance="default"
              level="1"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  undefined,
                ]
              }
            >
              <Text
                appearance="default"
                category="h3"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 30,
                      "fontWeight": "800",
                    },
                    Object {
                      "lineHeight": 60,
                      "textAlign": "center",
                    },
                  ]
                }
                testID="welcomeTitle"
              >
                Welcome to TiBoulot
              </Text>
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "marginBottom": 40,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ],
                  ]
                }
                testID="welcomeSubtitle"
              >
                Multiple services right in your pocket
              </Text>
            </View>
            <View
              appearance="default"
              level="1"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FE7439",
                    "borderColor": "#FE7439",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginBottom": 20,
                    "minHeight": 40,
                    "minWidth": 40,
                    "opacity": 1,
                    "paddingHorizontal": 10,
                    "paddingVertical": 12,
                  }
                }
                testID="signInBtn"
              >
                <Text
                  appearance="default"
                  category="p1"
                  style={
                    Array [
                      Object {
                        "color": "#222B45",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "400",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "marginHorizontal": 10,
                      },
                    ]
                  }
                >
                  Sign In
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(254, 116, 57, 0.08)",
                    "borderColor": "#FE7439",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginBottom": 20,
                    "minHeight": 40,
                    "minWidth": 40,
                    "opacity": 1,
                    "paddingHorizontal": 10,
                    "paddingVertical": 12,
                  }
                }
                testID="signUpBtn"
              >
                <Text
                  appearance="default"
                  category="p1"
                  style={
                    Array [
                      Object {
                        "color": "#222B45",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "400",
                      },
                      Object {
                        "color": "#FE7439",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "marginHorizontal": 10,
                      },
                    ]
                  }
                >
                  Sign Up
                </Text>
              </View>
            </View>
            <View
              appearance="default"
              level="1"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "display": "flex",
                    "flexDirection": "row",
                    "marginBottom": 40,
                    "marginVertical": 20,
                  },
                ]
              }
              testID="socialDivider"
            >
              <View
                appearance="default"
                style={
                  Array [
                    Object {
                      "backgroundColor": "#EDF1F7",
                      "height": 1,
                    },
                    Object {
                      "backgroundColor": "#CCC",
                      "flex": 1,
                    },
                  ]
                }
              />
              <Text
                appearance="default"
                category="c1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "flex": 1,
                        "textTransform": "lowercase",
                      },
                    ],
                  ]
                }
              >
                Or login using
              </Text>
              <View
                appearance="default"
                style={
                  Array [
                    Object {
                      "backgroundColor": "#EDF1F7",
                      "height": 1,
                    },
                    Object {
                      "backgroundColor": "#CCC",
                      "flex": 1,
                    },
                  ]
                }
              />
            </View>
            <View
              appearance="default"
              level="1"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4862A3",
                    "borderColor": "#4862A3",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                    "minHeight": 40,
                    "minWidth": 40,
                    "opacity": 1,
                    "paddingHorizontal": 10,
                    "paddingVertical": 12,
                    "width": 40,
                  }
                }
                testID="facebookBtn"
              >
                <View
                  style={
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={20}
                    bbWidth={20}
                    fill="#FFFFFF"
                    focusable={false}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {
                          "height": 20,
                          "marginHorizontal": 10,
                          "tintColor": "#FFFFFF",
                          "width": 20,
                        },
                        Object {
                          "flex": 0,
                          "height": 20,
                          "width": 20,
                        },
                      ]
                    }
                    vbHeight={24}
                    vbWidth={24}
                  >
                    <RNSVGGroup
                      fill={4294967295}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    >
                      <RNSVGGroup>
                        <RNSVGGroup>
                          <RNSVGRect
                            height="24"
                            matrix={
                              Array [
                                -1,
                                1.2246467991473532e-16,
                                -1.2246467991473532e-16,
                                -1,
                                24,
                                24,
                              ]
                            }
                            opacity={0}
                            width="24"
                            x={0}
                            y={0}
                          />
                          <RNSVGPath
                            d="M17 3.5a.5.5 0 0 0-.5-.5H14a4.77 4.77 0 0 0-5 4.5v2.7H6.5a.5.5 0 0 0-.5.5v2.6a.5.5 0 0 0 .5.5H9v6.7a.5.5 0 0 0 .5.5h3a.5.5 0 0 0 .5-.5v-6.7h2.62a.5.5 0 0 0 .49-.37l.72-2.6a.5.5 0 0 0-.48-.63H13V7.5a1 1 0 0 1 1-.9h2.5a.5.5 0 0 0 .5-.5z"
                          />
                        </RNSVGGroup>
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
